src/components/ui/Toast/Toast.tsx

```diff
@@ -1,35 +1,46 @@
-import Button from 'src/components/ui/Button'
-import { useEffect, useRef } from 'react'
+import { useEffect, useRef, useState } from 'react'
 import { useUI } from 'src/sdk/ui'
-
-const state = {
-  ERROR: 'bg-red',
-  WARNING: 'bg-yellow',
-  INFO: 'bg-green',
-}
+import Icon from 'src/components/ui/Icon'
 
 function Toast() {
   const { toasts, popToast } = useUI()
   const toast = toasts[toasts.length - 1]
-  const ref = useRef<NodeJS.Timeout[]>([])
+  const timeoutRef = useRef<NodeJS.Timeout>()
+
+  const [fade, setFade] = useState<'in' | 'out'>()
 
   useEffect(() => {
-    const id = setTimeout(() => {
-      popToast()
-    }, 2e3)
+    toast && setFade('in')
+  }, [toast])
+
+  useEffect(() => {
+    if (fade !== 'in') {
+      return
+    }
+
+    timeoutRef.current = setTimeout(() => setFade('out'), 2e3)
 
-    ref.current.push(id)
-  }, [popToast])
+    return () => timeoutRef.current && clearTimeout(timeoutRef.current)
+  }, [fade])
 
   if (toast === undefined) {
     return null
   }
 
   return (
-    <div>
-      <Button onClick={popToast}>Close</Button>
-      <div className={`h-36 text-white ${state[toast.status]}`}>
-        message: {toast.message}
+    <div
+      data-fs-toast
+      data-fs-toast-state={fade}
+      onAnimationEnd={() => fade === 'out' && popToast()}
```

Comment:
```suggestion
      onAnimationEnd={() => fade === 'out' && popToast()}
      role="status"
```

Reply:
We should improve this component's accessibility. I'll suggest the addition of role="status"

>  content is advisory information for the user but is not important enough to justify an [alert](https://www.w3.org/TR/wai-aria-1.1/#alert), often but not necessarily presented as a status bar.
- [reference](https://www.w3.org/TR/wai-aria-1.1/#status)

I think it's not important as an alert, but the designers can confirm that.